pkg/backend/httpstate/backend.go

```diff
@@ -762,38 +762,51 @@ func (b *cloudBackend) RemoveStack(ctx context.Context, stack backend.Stack, for
 	return b.client.DeleteStack(ctx, stackID, force)
 }
 
-func (b *cloudBackend) RenameStack(ctx context.Context, stack backend.Stack, newName tokens.QName) error {
+func (b *cloudBackend) RenameStack(ctx context.Context, stack backend.Stack,
+	newName tokens.QName) (backend.StackReference, error) {
 	stackID, err := b.getCloudStackIdentifier(stack.Ref())
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// Support a qualified stack name, which would also rename the stack's project too.
 	// e.g. if you want to change the project name on the Pulumi Console to reflect a
 	// new value in Pulumi.yaml.
 	newRef, err := b.ParseStackReference(string(newName))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	newIdentity, err := b.getCloudStackIdentifier(newRef)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if stackID.Owner != newIdentity.Owner {
-		url := "."
+		errMsg := "New stack owner, %s, does not match existing owner, %s.\n\n"
+
+		qn, _ := b.parseStackName(string(newName))
```

Comment:
It seems like we should delete lines 787-793.

Unless I'm missing something, `newName` doesn't get mutated after line 774, so there is no difference between `newRef` introduced earlier and `qn` defined here. Right?

Similarly, `cloudBackend::ParseStackReference` will always return the fully-qualified stack name (owner/project/name), so it should never be the case that `qn.Owner == ""`. (`cloudBackend::parseStackName` on the other hand will leave the value as "", since it doesn't infer missing values to fully-qualify the name.)

Reply:
> cloudBackend::parseStackName on the other hand will leave the value as "", since it doesn't infer missing values to fully-qualify the name.

Yes, that's exactly why we're using `parseStackName` here, so that we can check whether the owner was left off and give a better error message in that case ("did you forget?") -- because `ParseStackReference` populates it.

Reply:
BTW, the reason I made this change is that I found the current error message confusing. It seems more likely that I've simply forgotten to include the fully qualified name than it does I'm trying to transfer the stack between organizations. In the event I _am_ trying to transfer, the current error is great, but wanted to also cater to the likely common case too.